audio_service/lib/audio_service.dart

```diff
@@ -28,75 +28,111 @@ enum MediaButton {
 enum MediaAction {
   /// Stop playing audio.
   stop,
+  // 0
 
   /// Pause audio.
   pause,
+  // 1
 
   /// Play or resume audio.
   play,
+  // 2
 
   /// Rewind.
   rewind,
+  // 3
 
   /// Skip to the previous media item.
   skipToPrevious,
+  // 4
 
   /// Skip to the next media item.
   skipToNext,
+  // 5
 
   /// Fast forward.
   fastForward,
+  // 6
 
   /// Set a rating for the current media item.
   setRating,
+  // 7
 
   /// Seek within the current media item.
   seek,
+  // 8
 
   /// Toggle between playing and paused.
   playPause,
+  // 9
 
   /// Play a given media item by ID.
   playFromMediaId,
+  // 10
 
   /// Play media from a search.
   playFromSearch,
+  // 11
 
   /// Skip to a queue item.
   skipToQueueItem,
+  // 12
 
   /// Play media from a URI.
   playFromUri,
+  // 13
 
   /// Prepare media for playback.
   prepare,
+  // 14
 
   /// Prepare media for playback by ID.
   prepareFromMediaId,
+  // 15
 
   /// Prepare media for playback from a search.
   prepareFromSearch,
+  // 16
 
   /// Prepare media for playback from a URI.
   prepareFromUri,
+  // 17
 
   /// Set the repeat mode.
   setRepeatMode,
+  // 18
 
-  /// Unused.
-  unused_1,
+  /// Set captioning enabled.
+  setCaptioningEnabled,
```

Comment:
This shouldn't be changed because the position in the enum corresponds to Android's `PlaybackStateCompat.ACTION_*` constants.

Reply:
From the source:

```java
ACTION_PREPARE_FROM_URI = 1 << 17;
ACTION_SET_REPEAT_MODE = 1 << 18;
ACTION_SET_SHUFFLE_MODE_ENABLED = 1 << 19;
ACTION_SET_CAPTIONING_ENABLED = 1 << 20;
ACTION_SET_SHUFFLE_MODE = 1 << 21;
```

Reply:
I'm aware of this and confused - the values on enums seem to match up. Is there something that I've overlooked?

Reply:
They don't match up. You've got captioning coming right after repeat mode, but that's not what the source says.